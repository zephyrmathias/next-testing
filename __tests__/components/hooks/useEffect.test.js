import React from 'react'
import { shallow } from 'enzyme'
import { render } from '@testing-library/react'
import Example from '../../../components/hooks/useEffect'

describe('snapshot', () => {
  it('should match snapshot', () => {
    /**
     * shallow enzyme won't execute useEffect because
     * useEffect is different from componentDidMount
     *
     * https://reactjs.org/docs/hooks-reference.html#timing-of-effects
     * Unlike componentDidMount and componentDidUpdate,
     * the function passed to useEffect fires after layout and paint, during a deferred event.
     */
    const component = shallow(<Example />)
    expect(component).toMatchSnapshot()
  })
})

describe('useEffect', () => {
  // we use @testing-library/react or react-testing-library
  it('should contain "new title"', () => {
    const { container } = render(<Example />)
    const elem = container.querySelector('.title') // HTMLElement
    expect(elem.textContent).toBe('new title')
  })

  it('should match snapshot', () => {
    const { asFragment } = render(<Example />)
    expect(asFragment()).toMatchSnapshot()
  })
})
