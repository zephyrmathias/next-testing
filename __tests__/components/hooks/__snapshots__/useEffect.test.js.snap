// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should match snapshot 1`] = `
<div
  className="title"
>
  default title
</div>
`;

exports[`useEffect should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="title"
  >
    new title
  </div>
</DocumentFragment>
`;
